pkg/model/route.go

```diff
@@ -126,12 +131,97 @@ func (b *BackstageRoute) addToModel(model *BackstageModel, backstage bsv1.Backst
+func (b *BackstageRoute) updateAndValidate(model *BackstageModel, backstage bsv1.Backstage) error {
 	b.route.Spec.To.Name = model.backstageService.service.Name
+	b.updateAppConfigWithBaseUrls(model, backstage)
 	return nil
 }
 
 func (b *BackstageRoute) setMetaInfo(backstage bsv1.Backstage, scheme *runtime.Scheme) {
 	b.route.SetName(RouteName(backstage.Name))
 	setMetaInfo(b.route, backstage, scheme)
 }
+
+// updateAppConfigWithBaseUrls tries to set the baseUrl in the default app-config.
+// Note that this is purposely done on a best effort basis. So it is not considered an issue if the cluster ingress domain
+// could not be determined, since the user can always set it explicitly in their custom app-config.
+func (b *BackstageRoute) updateAppConfigWithBaseUrls(m *BackstageModel, backstage bsv1.Backstage) {
+	if m.appConfig == nil || m.appConfig.ConfigMap == nil {
+		klog.V(1).Infof(
+			"Default app-config ConfigMap not initialized yet - skipping automatic population of base URLS in the default app-config for Backstage %s",
+			backstage.Name)
+		return
+	}
+
+	baseUrl := buildBaseUrl(m, backstage)
+	updateFn := func(content string) (string, error) {
+		var appConfigData map[string]any
+		err := yaml.Unmarshal([]byte(content), &appConfigData)
+		if err != nil {
+			return "", fmt.Errorf("failed to decode app-config YAML: %w", err)
+		}
+		if appConfigData == nil {
+			appConfigData = make(map[string]any)
+		}
+		app, ok := appConfigData["app"].(map[string]any)
+		if !ok {
+			app = make(map[string]any)
+			appConfigData["app"] = app
+		}
+		app["baseUrl"] = baseUrl
+
+		backend, ok := appConfigData["backend"].(map[string]any)
+		if !ok {
+			backend = make(map[string]any)
+			appConfigData["backend"] = backend
+		}
+		backend["baseUrl"] = baseUrl
+
+		backendCors, ok := backend["cors"].(map[string]any)
+		if !ok {
+			backendCors = make(map[string]any)
+			backend["cors"] = backendCors
+		}
+		backendCors["origin"] = baseUrl
+
+		updated, err := yaml.Marshal(&appConfigData)
+		if err != nil {
+			return "", fmt.Errorf("failed to serialize updated app-config YAML: %w", err)
+		}
+		return string(updated), nil
+	}
+
+	for k, v := range m.appConfig.ConfigMap.Data {
+		updated, err := updateFn(v)
+		if err != nil {
+			klog.V(1).Infof("[warn] could not update base url in default app-config %q for backstage %s: %v",
+				k, backstage.Name, err)
+			continue
+		}
+		m.appConfig.ConfigMap.Data[k] = updated
+	}
+}
+
+// buildBaseUrl returns the base URL that should be considered as default on OpenShift,
+// per the cluster ingress domain and the Route spec.
+func buildBaseUrl(model *BackstageModel, backstage bsv1.Backstage) string {
+	host := fmt.Sprintf("%s-%s", RouteName(backstage.Name), backstage.Namespace)
+	appendIngressDomain := true
+	if backstage.Spec.Application != nil && backstage.Spec.Application.Route != nil {
+		// Per the Route spec, if a user specifies both the host and subdomain, the host takes precedence.
```

Comment:
Would it make sense to put this info to doc?

Reply:
Not sure... We are already mentioning in the [Route configuration docs](https://github.com/redhat-developer/rhdh-operator/blob/main/docs/configuration.md#route) that the Route fields allow specifying `some of the OpenShift Route specifications fields.`
So I think people can refer to the Route spec for further details.